### DISCUSSION TITLE
Sunbird-Saas - Exam Progress Loss on Page Refresh - Implementing Autosave and Seamless Resumption

### DISCUSSION THREAD

**Discussion by Pritams43** [Sunbird-Ed/Community#677] [Bugs]
**Issue:**
When taking an exam in the player, if the page is refreshed, the exam restarts from the beginning, causing answers to be lost.

**Desired Behavior:**
Upon refreshing the page during the exam, the user's answers should be saved, and the exam should resume from where it was left off, ensuring a seamless user experience. 

https://github.com/Sunbird-Ed/Community/assets/118895138/b146b3e8-44d3-4daf-a250-82d2dc1072a0

@vinukumar-vs @rajeevsathish @rajeshkumaravel @Krishnaj20

**Comment by sajeshkayyath** [comment]
@Pritams43 thanks for reaching out and suggesting this feature. 

There is no partial save functionality in the Player at the moment. The explicit browser refresh will reset it, which is the expected behaviour of the player.
Any specific use cases you are trying to handle by enabling this feature. Do let us know

cc @vinukumar-vs / @PoojaJManjooran

**Comment by anisha147** [comment]
Hi @Pritams43 

We hope your queries are answered. If yes, could you please close this thread from your end?

cc: @vinukumar-vs @sajeshkayyath
